src/scenes/Projects/Update/UpdateProjectDialog.tsx

```diff
@@ -0,0 +1,89 @@
+import { Typography } from '@material-ui/core';
+import React from 'react';
+import { Except } from 'type-fest';
+import { displayProjectStep, UpdateProjectInput } from '../../../api';
+import { projectStepToStatusMap } from '../../../api/projectSteps';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import { RadioField, RadioOption, SubmitError } from '../../../components/form';
+import { ProjectOverviewFragment } from '../Overview/ProjectOverview.generated';
+import { useUpdateProjectMutation } from './UpdateProject.generated';
+
+type UpdateProjectStepDialogProps = Except<
+  DialogFormProps<UpdateProjectInput>,
+  'onSubmit' | 'initialValues'
+> & {
+  project?: ProjectOverviewFragment;
+};
+
+export const UpdateProjectStepDialog = ({
+  project,
+  ...props
+}: UpdateProjectStepDialogProps) => {
+  const [updateProject] = useUpdateProjectMutation();
+
+  return (
+    <DialogForm<UpdateProjectInput>
+      title="Update Project Step"
+      closeLabel="Close"
+      submitLabel="Save"
+      onlyDirtySubmit
+      {...props}
+      initialValues={{
+        project: {
+          id: project?.id || '',
+          step: project?.step.value,
+        },
+      }}
+      onSubmit={async (input) => {
+        await updateProject({ variables: { input } });
+      }}
+    >
+      <SubmitError />
+      <RadioField name="project.step">
+        <Typography variant="h4">In Development</Typography>
+        {projectStepToStatusMap.InDevelopment.map((step) => (
```

Comment:
You could loop through the groups as well

Reply:
Fixed, much cleaner 👌